test/protobuf/protoc/generator_integration_test.exs

```diff
@@ -48,6 +48,30 @@ defmodule Protobuf.Protoc.GeneratorIntegrationTest do
     end)
   end
 
+  test "encoding map<string, string> validates the input" do
+    input =
+      My.Test.CreateAccountRequest.new!(%{
+        account: %{
+          instance_param_vals: %{
+            levels: %{
```

Comment:
You are nesting the map inside another field, it should be directly under `instance_param_vals` or you need to change the schema.

Reply:
@ericmj right, that is the issue, that we made the mistake of forgetting to marshall the `levels` (in this case) value to be a string rather than a map.

Notice that if I use the Protobuf serialize does work correctly, but if I use the JSON one, it doesn't.

Reply:
Can you fill out an issue that explains what happens now and what you would expect to happen?

Reply:
@ericmj yeah, for sure! I focused on replicating the issue to prove my hypothesis than explaining about. The reason why it is in Draft mode and left a comment in the description.

I will do it as soon as I find some time for it.

Reply:
We don't have any validation for JSON encoding other than for the `Google.Protobuf.` prefixed schemas so this may be a pretty big feature to implement. The best thing may be to just call `Protobuf.encode` before JSON encoding, but it's better to discuss that in an issue.

Reply:
> The best thing may be to just call Protobuf.encode before JSON encoding

Spot on, that is the question for sure here.

Do we want to do that?


Feels like a lot of extra overhead which is not a big deal for us, but for higher performance requirements may be suboptimal.

I am not sure.